authorization/invitation/service/invitation_service.go

```diff
@@ -162,3 +199,84 @@ func (s *invitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UU
 
 	return nil
 }
+
+type invitationNotification struct {
+	invitation *invitationrepo.Invitation
+	roles      []string
+}
+
+// processTeamInviteNotifications sends an e-mail notification to a user.
+func (s *invitationServiceImpl) processTeamInviteNotifications(ctx context.Context, team *account.Identity, inviterName string,
+	notifications []invitationNotification, witURL string) {
+	teamName := team.IdentityResource.Name
+
+	spaceName, err := lookupSpaceName(ctx, witURL, *team.IdentityResource.ParentResourceID)
+	if err != nil {
+		// What to do here?? just log an error and die?
+	}
+
+	var messages []notification.Message
+
+	for _, n := range notifications {
+		acceptURL := fmt.Sprintf("%s/api/invitations/accept?code=%s", s.config.GetAuthServiceURL(), n.invitation.AcceptToken.String())
+
+		messages = append(messages, notification.NewTeamInvitationEmail(n.invitation.Identity.UserID.UUID.String(),
+			teamName,
+			inviterName,
+			spaceName,
+			acceptURL))
+	}
+
+	s.Services().NotificationService().SendMessagesAsync(ctx, messages)
+}
+
+// processSpaceInviteNotifications sends an e-mail notification to a user.
+func (s *invitationServiceImpl) processSpaceInviteNotifications(ctx context.Context, space *resource.Resource,
+	inviterName string, notifications []invitationNotification, witURL string) {
+
+	spaceName, err := lookupSpaceName(ctx, witURL, space.ResourceID)
+	if err != nil {
+		// What to do here?? just log an error and die?
```

Comment:
Same comment as above.

Reply:
Since this is no longer running in a separate thread I now return an error